Topics/Tech_Stacks/GDB_Debug.md

```diff
@@ -0,0 +1,113 @@
+# Introduction how to debug by GDB
+
+## Table of contents
+
+#### What is GDB
+#### What can do by GDB
+#### How to use GDB
+
+
+## What is GDB?
+GDB (GNU Debugger) is a powerful open-source debugger commonly used in the field of computer science and software development, especially in the context of programming and debugging. 
+
+It is a command-line tool that allows developers to inspect and debug programs written in various programming languages, including C, C++, and even some other languages.
+
+Source: [GDB: The GNU Project Debugger](https://www.sourceware.org/gdb/)
+
+
+## What can do by GDB?
+
+1. **Start Your Program:**
+    GDB can start your program, specifying anything that might affect its behavior. This allows you to control the program's execution from the beginning.
+
+2. **Stop on Specified Conditions:**
+    You can set breakpoints in your code to make your program stop at specific conditions or events. Breakpoints can be set at particular line numbers, function names, or even based on variable values.
+
+2. **Examine Program State:**
+    While your program is paused at a breakpoint, GDB allows you to examine what has happened. You can inspect the values of variables, explore the call stack with a backtrace, and view the source code surrounding the current point of execution.
+
+3. **Modify Program State:**
+    GDB enables you to change things in your program, which is particularly useful for experimenting with bug fixes. You can modify variable values, continue program execution to observe different outcomes, and learn about multiple issues within your code.
+
+These features make GDB an essential tool for understanding and debugging programs, whether they are executing on the same machine as GDB (native), on another machine (remote), or on a simulator. GDB is compatible with most popular UNIX and Microsoft Windows variants, as well as macOS.
+
+
+
+## How to use GDB?
+
+1. **Install GDB:**
+    1. [Linux]: 
+        Debian-based systems like Ubuntu:`sudo apt-get install gdb`
+
+        Red Hat-based systems like Fedora: `sudo dnf install gdb`
+
+    2. [macOS]: `brew install gdb`
```

Comment:
I believe you need to install home-brew in order to run this. Maybe add a link to home-brew or GDB installation guide?

Reply:
Yeah, I will add a instruction to teach how to install `brew`.